Fix for issue #5594

dannyKBjj opened the pull request:
#### Pull Request (PR) description
The AppId was not being exported for MSFT_MicrosoftGGraphlistitem because the letter case for .appId was wrong ( the code was using appid instead of appId).

Corrected this. 

AppId now exports/test/sets/ etc. correctly for properties 
CompliantAppsList
AppsVisibilityList
AppsSingleAppModeList

#### This Pull Request (PR) fixes the following issues
    - Fixes #5594


#### Task list

<!--
    To aid community reviewers in reviewing and merging your PR, please take the time to run
    through the below checklist and make sure your PR has everything updated as required.

    Change to [x] for each task in the task list that applies to your PR. For those task that
    don't apply to you PR, leave those as is.
-->

- [x ] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [x ] Resource parameter descriptions added/updated in the schema.mof.
- [ x] Resource documentation added/updated in README.md.
- [ x] Resource settings.json file contains all required permissions.
- [ x] Examples appropriately added/updated.
- [x ] Unit tests added/updated.
- [ x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

dannyKBjj committed:
Fix for issue #5594

AppId not being exported for MSFT_MicrosoftGGraphlistitem because case for .appId in the code was wrong (appid instead of appId).

Corrected this. AppId now exports/test/sets/ etc. correctly for properties
CompliantAppsList
AppsVisibilityList
AppsSingleAppModeList

dannyKBjj committed:
Updated changelog.md

FabienTschanz commented:
@dannyKBjj Is it possible for you to not change the logic back to how it was previously now that we have the check for an export implemented? I guess simply updating the property names from `appid` to `appId` should be enough without the other changes. Does that work?

dannyKBjj commented:
Hi, i'm not sure what you mean. I thought that's basically all I did anyway? I see a lot of red in the pull request like I changed loads, but that really doesn't represent what I changed in the code!?

FabienTschanz commented:
I think the simplest thing is to just copy the red code and add it back into the resource, commit and push. That's how I would do it anyways, nothing fancy. Does that work for you if you update it?

dannyKBjj commented:
Yes, I'll do it on the weekend.

I think i see what's happened, I guess I branched off before the $Script:exportedInstance changes were made?

FabienTschanz commented:
> Yes, I'll do it on the weekend.
> 
> I think i see what's happened, I guess I branched off before the $Script:exportedInstance changes were made?

That's very likely, yes. Thanks a lot and sorry for the back and forth with those massive changes lately 😅

dannyKBjj committed:
Update MSFT_IntuneDeviceConfigurationPolicyiOS.psm1

made edits requested in comments

dannyKBjj commented:
following commit now seems to be showing the differences between this commit and the last commit :D

        if (-not $Script:exportedInstance) logic is now in.
        
        only changes between that and the current version is the appid values are now appId

FabienTschanz commented:
LGTM 🚀

dannyKBjj commented:
Hi, really sorry to bother you on this thread about another PR, but I've a few PR's that seem to have gone dead. 

#5588 - made requested changes last week, no feedback.
#5640 - unit tests work fine in my environment, but not when I initiate the pull request. Not a clue what the issue is?
#5673 - again, unit tests playing up. Work fine in my environment so it isn't the module, or the test script?

Sorry if I'm being annoying, just quite keen to get these merged :-D

ykuijs reviewed (changes requested):
Please make sure you update with Dev. That contains updates to the changelog.

After that make sure your entry is created under the Unreleased section in that version of the changelog.

ykuijs commented:
> Hi, really sorry to bother you on this thread about another PR, but I've a few PR's that seem to have gone dead.
> 
> #5588 - made requested changes last week, no feedback. #5640 - unit tests work fine in my environment, but not when I initiate the pull request. Not a clue what the issue is? #5673 - again, unit tests playing up. Work fine in my environment so it isn't the module, or the test script?
> 
> Sorry if I'm being annoying, just quite keen to get these merged :-D

Sorry, have been a little busy the last few days with very little time to review PRs.
